### PR TITLE
tests(patch) added tests for core patches

### PR DESCRIPTION
**Pull request by dndx**
Not everything makes sense to test. This PR tests majority of the patches we apply to NGINX and OpenResty core.

This PR depends on `lua.openssl` library to run.

### CODE REVIEW

**Comment by hishamhm** [comment]
I think it would be important to make these tests run in this repo's CI, otherwise they may end up sitting in `t/` forgotten, in case changes are made to the patches.

File: t/001-balancer-ssl-ctx.t [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+--- error_log
+SSL_do_handshake() failed (SSL: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed)
+--- no_error_log
+[warn]
+[error]
+
+
+
+=== TEST 3: set SSL_CTX for non-https upstream
+--- SKIP
```
Comment (thibaultcha) [inline t/001-balancer-ssl-ctx.t:105]: Could we add a reason for the skip for readers/future selves?

**Review by thibaultcha — APPROVED** [review]
:+1:

**Comment by dndx** [comment]
> I think it would be important to make these tests run in this repo's CI, otherwise they may end up sitting in `t/` forgotten, in case changes are made to the patches.

Yes, we will run those alongside with the rest of the `ngx_lua` and `stream_lua` tests in a separate PR.

**Pull request merged**